pkg/controller/utils/utils.go

```diff
@@ -184,16 +184,21 @@ func CheckLicenseKey(ctx context.Context, cli client.Client) error {
 	return cli.Get(ctx, DefaultInstanceKey, instance)
 }
 
-// ValidateCertPair checks if the given secret exists and if so
-// that it contains key and cert fields. If a secret exists then it is returned.
+// ValidateCertPair validates the cert pair secret in the tigera-operator namespace.
+func ValidateCertPair(client client.Client, certPairSecretName, keyName, certName string) (*corev1.Secret, error) {
+	return ValidateCertPairInNamespace(client, render.OperatorNamespace(), certPairSecretName, keyName, certName)
+}
+
+// ValidateCertPairInNamespace checks if the given secret exists in the given
+// namespace and if so that it contains key and cert fields. If a secret exists then it is returned.
 // If there is an error accessing the secret (except NotFound) or the cert
 // does not have both a key and cert field then an appropriate error is returned.
 // If no secret exists then nil, nil is returned to represent that no cert is valid.
-func ValidateCertPair(client client.Client, certPairSecretName, keyName, certName string) (*corev1.Secret, error) {
+func ValidateCertPairInNamespace(client client.Client, namespace, certPairSecretName, keyName, certName string) (*corev1.Secret, error) {
```

Comment:
Maybe we should add a note or TODO saying this is a temporary function later we should change ValidateCertPair to accept a namespace.